provisioning/transport/amqp/src/AmqpClientSession.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Amqp.Framing;
+
+namespace Microsoft.Azure.Devices.Provisioning.Client.Transport.Amqp
+{
+    public class AmqpClientSession
+    {
+        readonly AmqpClientConnection amqpConnection;
+
+        public AmqpClientSession(AmqpClientConnection amqpTestConnection)
+        {
+            this.amqpConnection = amqpTestConnection;
+            AmqpSessionSettings = new AmqpSessionSettings();
+        }
+
+        private bool _isSessionClosed;
+
+        internal AmqpSession AmqpSession { get; private set; }
+
+        public AmqpSessionSettings AmqpSessionSettings { get; private set; }
+
+        public AmqpClientLink SendingLink { get; private set; }
+
+        public AmqpClientLink ReceivingLink { get; private set; }
+
+        public bool IsSessionClosed => _isSessionClosed;
+
+        public async Task OpenAsync(TimeSpan timeout)
+        {
+            // Create the Session
+            var amqpTestLinkFactory = new AmqpLinkFactory();
+            amqpTestLinkFactory.LinkCreated += this.OnLinkCreated;
+            this.AmqpSession = new AmqpSession(this.amqpConnection.AmqpConnection, this.AmqpSessionSettings, amqpTestLinkFactory);
+            this.amqpConnection.AmqpConnection.AddSession(this.AmqpSession, new ushort?());
```

Comment:
I believe that `default(ushort?)` is `null`. Since we're not passing an actual value, we could just pass null for the channel?